libs/ml/include/ml/core/node.hpp

```diff
@@ -141,6 +143,105 @@ class Node
     return static_cast<bool>(cached_output_status_ == CachedOutputState::VALID_CACHE);
   }
 
+  virtual uint64_t ForwardPassChargeCost()
+  {
+    static constexpr uint64_t my_default_cost = 100;  // TODO(VH): Calculate/measure me
+    // TODO(VH): call OpForwardCost() of the Op - we need to know Data shape to do this.
+    if (input_nodes_.empty())
+    {
+      FETCH_LOG_INFO("Node", " Input node reached: default cost " +
+                                 std::to_string(my_default_cost) + " added.");
+      return my_default_cost;
+    }
+    uint64_t                           total_cost = my_default_cost;
+    std::vector<std::vector<uint64_t>> input_shapes;
+    for (auto const &i : input_nodes_)
+    {
+      if (auto input_node_ptr = i.lock())
+      {
+        auto const input_node_forward_cost = input_node_ptr->ForwardPassChargeCost();
+        total_cost += input_node_forward_cost;
```

Comment:
why not just:
total_cost += input_node_ptr->ForwardPassChargeCost();

Reply:
done